nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/test/java/org/apache/nifi/jasn1/JASN1ReaderTest.java

```diff
@@ -78,4 +89,82 @@ public void testCanLoadClassCompiledFromAsn() throws Exception {
         assertEquals("org.apache.nifi.jasn1.test.RootType", actualRootModelName);
         assertNotNull(actual);
     }
+
+    @Test
+    public void testAsnFileDoesntExist() throws Exception {
+        // GIVEN
+        ConfigurationContext context = mock(ConfigurationContext.class, RETURNS_DEEP_STUBS);
+        when(context.getProperty(ASN_FILES).isSet()).thenReturn(true);
+        when(context.getProperty(ASN_FILES).evaluateAttributeExpressions().getValue()).thenReturn(
+                "src/test/resources/test.asn,src/test/resources/doesnt_exist.asn"
+        );
+
+        // WHEN
+        ProcessException processException = assertThrows(
+                ProcessException.class,
+                () -> testSubject.onEnabled(context)
+        );
+        Throwable cause = processException.getCause();
+
+        assertEquals(FileNotFoundException.class, cause.getClass());
+        assertThat(cause.getMessage(), containsString("src/test/resources/doesnt_exist.asn"));
+    }
+
+    @Test
+    public void testCantParseAsn() throws Exception {
+        // GIVEN
+        String asnFiles = "src/test/resources/cant_parse.asn";
+
+        List<String> expectedErrorMessages = Arrays.asList(
+                "line 11:5: unexpected token: field3",
+                "line 17:33: unexpected token: ["
+        );
+
+        // WHEN
+        // THEN
+        testError(asnFiles, expectedErrorMessages);
+    }
+
+    @Test
+    public void testCantCompileAsn() throws Exception {
+        // GIVEN
+        String asnFiles = "src/test/resources/cant_compile.asn";
+
+        List<String> expectedErrorMessages = Arrays.asList(
+                "class SAMENAMEWithDifferentCase is public, should be declared in a file named SAMENAMEWithDifferentCase.java",
+                "cannot find symbol\n" +
+                        "  symbol:   class SameNameWithDifferentCase\n" +
+                        "  location: class org.apache.nifi.jasn1.test.SAMENAMEWithDifferentCase",
+                "incompatible types: com.beanit.asn1bean.ber.types.BerInteger cannot be converted to com.beanit.asn1bean.ber.BerLength",
+                "incompatible types: boolean cannot be converted to java.io.OutputStream",
+                "Some messages have been simplified; recompile with -Xdiags:verbose to get full output"
+        );
```

Comment:
Expecting precise error messages makes tests more brittle, and appears to introduce problems with different language locales. Recommend asserting particular keywords, such as a class name, instead of full messages.

Reply:
The error messages by the underlying libraries are very technical and don't convey the root cause very well if at all. So there are no good keywords the presence/lack of which would correspond to a given error.

For the same reason I documented the errors exactly as they appear in the additionalDetails.html and provided an explanation for them.

I expect the list of known issues to grow but not the existing ones to change as the underlying apis don't change often. And when they do, we need to make sure we understand the change.

However I'll update the tests to mention the additionalDetails. This way any change will be captured by the tests and hopefully the user won't get stuck with outdated info either.